ocsp.go

```diff
@@ -382,21 +490,21 @@ func getRevocationStatus(wg *sync.WaitGroup, ocspStatusChan chan<- *ocspStatus,
 	headers["Host"] = u.Hostname()
 	ocspRes, ocspResBytes, ocspS := retryOCSP(ocspClient, http.NewRequest, ocspHost, headers, ocspReq, issuer, retryOCSPTimeout, retryOCSPHTTPTimeout)
 	if ocspS.code != ocspSuccess {
-		ocspStatusChan <- ocspS
-		return
+		return ocspS
 	}
-	encodedCertIDBase64 := base64.StdEncoding.EncodeToString(encodedCertID)
-	ocspStatusChan <- validateOCSP(encodedCertIDBase64, ocspRes, subject)
+
+	ret := validateOCSP(encodedCertID, ocspRes, subject)
 	v := []interface{}{float64(time.Now().UTC().Unix()), base64.StdEncoding.EncodeToString(ocspResBytes)}
 	ocspResponseCacheLock.Lock()
-	ocspResponseCache[encodedCertIDBase64] = v
+	ocspResponseCache[*encodedCertID] = v
```

Comment:
We should ideally check the ret value before updating the cache with the response. If the response is invalid then in this case we would be updating the cache with an invalid entry.

Reply:
ok. will do.